tools/chainsync-loadtest/src/fetch_chain.rs

```diff
@@ -0,0 +1,68 @@
+use std::sync::Arc;
+
+use log::info;
+
+use crate::concurrency::{Ctx, Scope};
+use crate::network;
+use anyhow::Context;
+use tokio::time;
+
+use near_primitives::hash::CryptoHash;
+
+pub async fn run(
+    ctx: Ctx,
+    network: Arc<network::Network>,
+    start_block_hash: CryptoHash,
+) -> anyhow::Result<()> {
+    info!("SYNC start");
+    let peers = network.info(&ctx).await?;
+    let target_height = peers.highest_height_peers[0].chain_info.height;
+    info!("SYNC target_height = {}", target_height);
+    Scope::run(&ctx, |ctx, s| async move {
+        s.spawn_weak({
```

Comment:
what does spawn_weak do? do we have to clean it up/stop somehow afterwards?

Reply:
PTAL at the extensive comment in the scope.rs.
Scope awaits for all routines to complete before the scope is completed.
If any routine returns an error, the whole scope is cancelled, and cancellation is propagated to all other routines of the scope.
Still there are 2 kinds of routines in the scope: "main" and "weak". As soon as all "main" routines complete, the scope gets cancelled (even if there was no error), and the cancellation is propagated to the remaining ("weak") routines. The idea is that "weak" routines are not expected to ever complete on their own.